sdcm/cluster.py

```diff
@@ -2434,111 +2443,226 @@ def check_node_health(self):
+        peers_details = {}
+        for line in cql_result[3:-2]:
+            line_splitted = line.split('|')
+            peers_details[line_splitted[0].strip()] = {'data_center': line_splitted[1].strip(),
+                                                       'host_id': line_splitted[2].strip(),
+                                                       'rack': line_splitted[3].strip(),
+                                                       'release_version': line_splitted[4].strip(),
+                                                       'rpc_address': line_splitted[5].strip(),
+                                                       'schema_version': line_splitted[6].strip(),
+                                                       'supported_features': line_splitted[7].strip()}
+
+        return peers_details
+
+    @retrying(n=5, sleep_time=10, raise_on_exceeded=False)
+    def get_gossip_info(self):
+        gossip_info = self.run_nodetool('gossipinfo', verbose=False)
         gossip_node_schemas = {}
-        schema, ip, status = '', '', ''
+        schema, ip, status, dc = '', '', '', ''
         for line in gossip_info.stdout.split():
             if line.startswith('SCHEMA:'):
                 schema = line.replace('SCHEMA:', '')
             elif line.startswith('RPC_ADDRESS:'):
                 ip = line.replace('RPC_ADDRESS:', '')
             elif line.startswith('STATUS:'):
                 status = line.replace('STATUS:', '').split(',')[0]
-            # STATUS is "LEFT": in case the node was decommissioned
-            # STATUS is "removed": in case the node was removed by nodetool removenode
-            # STATUS is "BOOT": node during boot and not exists in the cluster yet
-            # and they will remain in the gossipinfo 3 days.
-            # It's expected behaviour and we won't send the error in this case
+            elif line.startswith('DC:'):
+                dc = line.replace('DC:', '').split(',')[0]
+
             if schema and ip and status:
-                if status not in ['LEFT', 'removed', 'BOOT', 'shutdown']:
-                    gossip_node_schemas[ip] = {'schema': schema, 'status': status}
-                schema, ip, status = '', '', ''
+                gossip_node_schemas[ip] = {'schema': schema, 'status': status, 'dc': dc}
+                schema, ip, status, dc = '', '', '', ''
 
-        self.log.info('Gossipinfo schema version and status of all nodes: {}'.format(gossip_node_schemas))
+        return gossip_node_schemas
 
-        # Validate that ALL initiated nodes in the gossip
-        cluster_nodes = [node.ip_address for node in self.parent_cluster.nodes if node.db_init_finished]
+    def if_node_is_target_node(self, node_ip):
+        for node in self.parent_cluster.nodes:
+            if node.ip_address == node_ip:
+                return f' ({node.running_nemesis} nemesis target node)' if node.running_nemesis else ' (not target node)'
+        return ''
```

Comment:
It's really unclear from the name what this method returns.

I can suggest to add a method `get_node_by_ip(node_ip: str) -> Optional[BaseNode]` to `BaseCluster` class,
and create separate helper function/method to format string as you do in your code (`"{ip}{is_target}"`})

Reply:
Fixed